ecc/bls12-377/fr/iop/utils_test.go

```diff
@@ -0,0 +1,784 @@
+// a generateTestPolynomials
+func getListInfo() []Form {
+
+	res := make([]Form, 12)
+
+	res[0].Basis = Canonical
+	res[0].Layout = Regular
+	res[0].Status = Locked
+
+	res[1].Basis = Canonical
+	res[1].Layout = Regular
+	res[1].Status = Unlocked
+
+	res[2].Basis = Canonical
+	res[2].Layout = BitReverse
+	res[2].Status = Locked
+
+	res[3].Basis = Canonical
+	res[3].Layout = BitReverse
+	res[3].Status = Unlocked
+
+	res[4].Basis = Lagrange
+	res[4].Layout = Regular
+	res[4].Status = Locked
+
+	res[5].Basis = Lagrange
+	res[5].Layout = Regular
+	res[5].Status = Unlocked
+
+	res[6].Basis = Lagrange
+	res[6].Layout = BitReverse
+	res[6].Status = Locked
+
+	res[7].Basis = Lagrange
+	res[7].Layout = BitReverse
+	res[7].Status = Unlocked
+
+	res[8].Basis = LagrangeCoset
+	res[8].Layout = Regular
+	res[8].Status = Locked
+
+	res[9].Basis = LagrangeCoset
+	res[9].Layout = Regular
+	res[9].Status = Unlocked
+
+	res[10].Basis = LagrangeCoset
+	res[10].Layout = BitReverse
+	res[10].Status = Locked
+
+	res[11].Basis = LagrangeCoset
+	res[11].Layout = BitReverse
+	res[11].Status = Unlocked
+
+	return res
+}
+
+func getCopy(l []Polynomial) []Polynomial {
+	r := make([]Polynomial, len(l))
+	for i := 0; i < len(l); i++ {
+		r[i].Coefficients = make([]fr.Element, len(l[i].Coefficients))
+		copy(r[i].Coefficients, l[i].Coefficients)
+		r[i].Info = l[i].Info
+	}
+	return r
+}
+
+func cmpCoefficents(p, q []fr.Element) bool {
+	if len(p) != len(q) {
+		return false
+	}
+	res := true
+	for i := 0; i < len(p); i++ {
+		res = res && (p[i].Equal(&q[i]))
+	}
+	return res
+}
+
+func TestPutInLagrangeForm(t *testing.T) {
+
+	size := 64
+	domain := fft.NewDomain(uint64(size))
+
+	// reference vector in Lagrange-regular form
+	c := randomVector(size)
+	var regular, bitreversed Polynomial
+	regular.Coefficients = c
+	regular.Info.Basis = Lagrange
+	regular.Info.Layout = Regular
+	regular.Info.Status = Locked
+	bitreversed.Info = regular.Info
+	bitreversed.Coefficients = make([]fr.Element, size)
+	copy(bitreversed.Coefficients, regular.Coefficients)
+	polynomials := make([]Polynomial, 12)
+	for i := 0; i < 12; i++ {
+		polynomials[i] = *fromLagrange[i](&regular, domain)
+	}
+
+	// check that the info field is filled correctly
+	info := getListInfo()
+	for i := 0; i < 12; i++ {
+		if polynomials[i].Info != info[i] {
+			t.Fatal("info field is not filled correcly")
+		}
+	}
+
+	// check that r has not been modified
+	if !reflect.DeepEqual(&regular, &bitreversed) {
+		t.Fatal("reference polynomial should not be modified")
+	}
+
+	// bit the reference vector for the bit reversed case
+	bitreversed.Info.Layout = BitReverse
+	fft.BitReverse(bitreversed.Coefficients)
+
+	// get a backup
+	backupPolynomials := getCopy(polynomials)
+
+	// create the Lagrange form...
+	lagrangePolynomials := make([]Polynomial, 12)
+	for i := 0; i < 12; i++ {
+		lagrangePolynomials[i] = *toLagrange(&polynomials[i], domain)
+	}
+
+	// compare the results that should be in regular form
```

Comment:
I don't know how to review that ^^

Reply:
yeah no worries those are just utility functions that I checked with Sage